internal/k8s/controllers/config_controller.go

```diff
@@ -149,6 +158,28 @@ func (r *ConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 }
 
 func (r *ConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	p := predicate.NewPredicateFuncs(
+		func(obj client.Object) bool {
+			newNodeObj, ok := obj.(*corev1.Node)
+			if !ok {
+				return true
+			}
+			err := r.Get(context.Background(), types.NamespacedName{Namespace: newNodeObj.Namespace,
+				Name: newNodeObj.Name}, &corev1.Node{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return true
+				}
+				level.Error(r.Logger).Log("controller", "ConfigReconciler", "error", "failed to get node", "error", err)
+				return false
+			}
+			if existingNodeObj, ok := r.Nodes[newNodeObj.Name]; ok &&
+				labels.Set(existingNodeObj.Labels).String() == labels.Set(newNodeObj.Labels).String() {
```

Comment:
will these be always in the right order?

Reply:
yes, changed to use `labels.Equals` method.